src/JitInterface/src/SystemVStructClassificator.cs

```diff
@@ -0,0 +1,648 @@
+                return SystemVClassificationType.SystemVClassificationTypeIntegerReference;
+
+            case SystemVClassificationType.SystemVClassificationTypeIntegerByRef:
+                // IntegerByReference can only merge with IntegerByReference.
+                Debug.Assert(originalClassification == SystemVClassificationType.SystemVClassificationTypeIntegerByRef);
+                return SystemVClassificationType.SystemVClassificationTypeIntegerByRef;
+
+            default:
+                Debug.Assert(false); // Unexpected type.
+                return SystemVClassificationType.SystemVClassificationTypeUnknown;
+            }
+        }
+
+        /// <summary>
+        /// Returns 'true' if the struct is passed in registers, 'false' otherwise.
+        /// </summary>
+        private static bool ClassifyEightBytes(TypeDesc typeDesc, 
+                                               ref SystemVStructRegisterPassingHelper helper,
+                                               int startOffsetOfStruct)
+        {
+            FieldDesc firstField = null;
+            int numIntroducedFields = 0;
+            foreach (FieldDesc field in typeDesc.GetFields())
+            {
+                if (!field.IsLiteral && !field.IsStatic)
+                {
+                    if (firstField == null)
+                    {
+                        firstField = field;
+                    }
+                    numIntroducedFields++;
+                }
+            }
+
+            if (numIntroducedFields == 0)
+            {
+                return false;
+            }
+
+            // The SIMD Intrinsic types are meant to be handled specially and should not be passed as struct registers
+            if (typeDesc.IsIntrinsic)
+            {
+                InstantiatedType instantiatedType = typeDesc as InstantiatedType;
+                if (instantiatedType != null)
+                {
+                    string typeName = instantiatedType.Name;
+                    string namespaceName = instantiatedType.Namespace;
+
+                    // TODO: is this what we get for the typeName?
+                    if (typeName == "Vector256`1" || typeName == "Vector128`1" || typeName == "Vector64`1")
+                    {
+                        Debug.Assert(namespaceName == "System.Runtime.Intrinsics");
+                        return false;
+                    }
+
+                    if ((typeName ==  "Vector`1") && (namespaceName == "System.Numerics"))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            MetadataType mdType = typeDesc as MetadataType;
+            Debug.Assert(mdType != null);
+
+            TypeDesc firstFieldElementType = firstField.FieldType;
+            int firstFieldSize = firstFieldElementType.GetElementSize().AsInt;
+
+            bool isFixedBuffer = mdType.HasCustomAttribute("System.Runtime.CompilerServices", "FixedBufferAttribute");
```

Comment:
All the code on the CoreCLR side was just to avoid looking at the attribute. It is ugly and as the comment said, not 100% reliable. That's why I have replaced it by the simple attribute check. Or can you see a case when a type would be a fixed buffer and not have the attribute?

Reply:
One can put the attribute on anything. It would be annoying if it resulted in a data corruption only when precompiling - it's better to match what CoreCLR does. From a quick look CoreCLR doesn't use the attribute to decide anything.

On a second read, the attribute check is not correct anyway - the attribute gets applied on the field, not on the field type - maybe we have a test hole if this didn't trigger any failures.